### DISCUSSION TITLE
Silly question: Sanitize vs Validate. What is the difference ?

### DISCUSSION THREAD

**Discussion by a3an-de** [thombashi/pathvalidate#32] [Q&A] [answered]
Sanitize vs Validate. What is the difference ?

**Comment by thombashi** [comment, marked as answer]
In the `pathvalidate` package, `sanitize` means that remove/replace invalid characters from a value to make a valid filename/filepath for a specific platform.
`validate` means checking whether invalid characters are included in the value. And this has not changed the original value.

**Comment by impredicative** [comment]
`sanitize` will modify as needed, and return the sanitary form.

`validate` will not modify; it will raise an exception if the validation fails.

`is_valid` will return whether valid or not.
